experimental/FTheoryTools/src/AbstractFTheoryModels/attributes.jl

```diff
@@ -1096,6 +1096,134 @@ function global_gauge_quotients(m::AbstractFTheoryModel)
 end
 
 
+@doc raw"""
+    chern_class_c1(m::AbstractFTheoryModel)
+
+If the elliptically fibered n-fold $Y_n$ underlying the F-theory model in question is given
+as a hypersurface in a toric ambient space, we can compute a cohomology class $h$ on the
+toric ambient space $X_\Sigma$, such that its restriction to $Y_n$ is the first Chern class
+$c_1$ of the tangent bundle of $Y_n$. If those assumptions are satisfied, this method returns
+this very cohomology class $h$, otherwise it raises and error.
+
+Note that $Y_n$ is a Calabi-Yau variety precisely if $c_1$ is trivial. Thus, the restriction
+of $h$ to the hypersurface $Y_n$ must be trivial. Upon closer inspection, in the given toric
+setting, this is equivalent to $h$ being trivial.
+
+The computation of the cohomology ring verifies if the toric variety is simplicial and
+complete. The check for it to be complete can be very time consuming. This can be switched
+off by setting the optional argument `check` to the value `falue`, as in the example below.
+
+```jldoctest
+julia> qsm_model = literature_model(arxiv_id = "1903.00009", model_parameters = Dict("k" => 4))
+Hypersurface model over a concrete base
+
+julia> h = chern_class_c1(qsm_model; check = false)
+Cohomology class on a normal toric variety given by 0
+
+julia> is_trivial(h)
+true
+```
+"""
+function chern_class_c1(m::AbstractFTheoryModel; check::Bool = true)
+  @req (m isa WeierstrassModel || m isa GlobalTateModel || m isa HypersurfaceModel) "First Chern class of F-theory model supported for Weierstrass, global Tate and hypersurface models only"
+  @req base_space(m) isa NormalToricVariety "First Chern class of F-theory model currently supported only for toric base"
+  @req ambient_space(m) isa NormalToricVariety "Firfalsest Chern class of F-theory model currently supported only for toric ambient space"
+
+  # Check if the answer is known
+  if has_attribute(m, :chern_class_c1)
+    return get_attribute(m, :chern_class_c1)
+  end
```

Comment:
Maybe you could just do
```suggestion
@attr SOME_TYPE function chern_class_c1(m::AbstractFTheoryModel; check::Bool = true)
  @req (m isa WeierstrassModel || m isa GlobalTateModel || m isa HypersurfaceModel) "First Chern class of F-theory model supported for Weierstrass, global Tate and hypersurface models only"
  @req base_space(m) isa NormalToricVariety "First Chern class of F-theory model currently supported only for toric base"
  @req ambient_space(m) isa NormalToricVariety "Firfalsest Chern class of F-theory model currently supported only for toric ambient space"

```
but possibly this does not work since there is an optional argument...

Reply:
Yes, the current @attr macro expects a unary function and is not happy about the optional argument. This could of course be improved in the future, but for now I am happy that this code works as expected.